challenge.rb

```diff
@@ -1,26 +1,40 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  input.collect do |x|
+    x.capitalize
+  end
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  input.select do |x|
+    x == "dog"
+  end
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+  input.reject do |x|
+    x == "dog"
+  end
 end
 
 def count_the_animals(input)
   #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+  input.find_all do |x|
+  x[0] == "c" || x[0] == "d"
```

Comment:
Would you believe Ruby has a method on String for just this?
http://ruby-doc.org/core-2.0.0/String.html#method-i-start_with-3F

It is convention to indent the content of the block (this one line) by two more spaces.